docker-images/build.sh

```diff
@@ -63,20 +65,67 @@ function build {
     # Images depending on Kafka version (possibly indirectly thru FROM)
     for kafka_version in "${!version_checksums[@]}"
     do
-        sha=${version_checksums[$kafka_version]}
+        echo "Creating container image for Kafka $kafka_version"
+
+        expected_sha=${version_checksums[$kafka_version]}
         lib_directory=${version_libs[$kafka_version]}
+
+        binary_file_dir="./$kafka_image/tmp"
+
+        # If there is a file specified for this version of Kafka use that instead of the specified URL
+        if [ ${version_binary_files[$kafka_version]} ]
+        then 
+
+            custom_binary_file_path=${version_binary_files[$kafka_version]}
+            binary_file_name=$(basename "$custom_binary_file_path")
+            binary_file_path="$binary_file_dir/$binary_file_name"
+            echo "Kafka binary file field was specified, using $custom_binary_file_path instead of URL"
+
+            if [ -f "$binary_file_path" ]
+            then
+                echo "$binary_file_name already present in build directory"
+            else
+                echo "Copying $binary_file_name to build directory"
+                cp "$custom_binary_file_path" "$binary_file_path" 
+            fi
+            
+        elif [ ${version_binary_urls[$kafka_version]} ]
+        then
+
+            binary_file_url=${version_binary_urls[$kafka_version]}
+            binary_file_name=$(basename "$binary_file_url")
+            binary_file_path="$binary_file_dir/$binary_file_name"
+
+            if [ -f "$binary_file_path" ]
+            then
+                echo "$binary_file_name already present in build directory"
+            else
+                echo "Downloading Kafka $kafka_version binaries from: $binary_file_url"
+                curl --output "$binary_file_path" "$binary_file_url"
+            fi
+
+        fi
+            
+        # Do the checksum on the Kafka binaries
+        kafka_checksum="$expected_sha $binary_file_path"
+        kafka_checksum_filepath="$binary_file_path.sha512"
+        echo "$kafka_checksum" > "$kafka_checksum_filepath"
+        sha512sum --check "$kafka_checksum_filepath"
+
+        relative_binary_file_path="./tmp/$binary_file_name"
```

Comment:
You seem to pull a lot of stuff locally and keeping them as cache for next time (which is good). But I do not see anything for `make clean` to delete these files. Are these directories already used and deleted? Or do we nee dto add it?

Reply:
The binaries are stored under the `tmp` directory in `kafka` which is already `rm -rf`'d by the `clean` command in `docker-images/kafka/Makefile`.